compiler/src/test/kotlin/symboltest.kt

```diff
@@ -0,0 +1,145 @@
+                is BoolLiteral -> assertEquals(value, literal.value)
+                is FloatLiteral -> assertEquals(value, literal.value)
+                else -> throw AssertionFailedError("Type was not of assignable type! Was " + literal.getType())
+            }
+        } catch (e: AssertionFailedError) { // If any assertion errors thrown, catch and return false
+            return false
+        } catch (e: NullPointerException) { // Catch NPE's which may be thrown due to looking up ident as key
+            e.printStackTrace()
+            return false
+        }
+        return true
+    }
+
+    /**
+     * Dispatches assignment statement tests
+     */
+    @Test
+    fun assignStmtTest() {
+        for (i in -50..50) assertTrue(assignStmtPass("x", i))
+        assertTrue(assignStmtPass("floatingPoint", 1636098.1239487.toFloat()))
+        assertTrue(assignStmtPass("falseBool", false))
+        assertTrue(assignStmtPass("trueBool", true))
+    }
+
+    /**
+     * Tests whether parser recognises syntax errors on reserved words from symbol table
+     */
+    @Test
+    fun parserReservedSymbolsTest() {
+        RESERVED_WORDS.forEach {
+            val parser = getParser(getWorldDecl() + getStateDecl(ident = it))
+            // Stop parser from printing errors to stderr for less noisy console
+            parser.removeErrorListeners()
+            // Parse program
+            parser.start()
+
+            // Assert that syntax-errors are recognised for reserved words
+            assertTrue(parser.numberOfSyntaxErrors > 0, "Test failed on: $it")
+        }
+    }
+
+    /**
+     * Tests whether each error on compiled program with multiple assignments on same identifier registers an
+     * SymbolRedefinitionError and whether the identifier is equal to passed parameter.
+     * Can take a generic list of values, which should be limited to int, float, and boolean types,
+     * as language only supports those literals
+     *
+     * @param ident Identifier to use for each assignment statement
+     * @param values List of values which will be assigned to given identifier
+     */
+    fun <T> symbolRedefinition(ident: String, values: List<T>): Boolean {
+        val stringBuilder = StringBuilder()
+        // For each value, create assignment expression with equal identifier and given value
+        values.forEach {
+            stringBuilder.appendln("let $ident = ${it.toString()};")
+        }
+
+        // Compile boilerplate program with state having the constructed body
+        compileProgram(getWorldDecl() + getStateDecl(body = stringBuilder.toString()))
+
+        try {
+            // Assert that errors are registered
+            assertTrue(ErrorLogger.hasErrors())
+
+            // For each error recorded, assert that error is of SymbolRedefinitionError-type and with given identifier
+            ErrorLogger.allErrors().forEach {
+                assertTrue(it is SymbolRedefinitionError, "Class assertion error at: $it")
```

Comment:
I don't understand the error message on the assertion, can you elaborate?

Reply:
Elaborated on recent commit. Resolve if sufficient.

Reply:
Relevant commit: a0838985ec3bac55d93cd1ff3972db1871779684

Reply:
Please don't use a shorthand for class names

Reply:
You're right, fixed in: 3cacee92291532c9185a91e024ec3f3764de424e.